Samples/WinMLSamplesGallery/WinMLSamplesGallery/Samples/Batching/Batching.xaml.cs

```diff
@@ -0,0 +1,240 @@
+using System;
+using System.Collections.Generic;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Windows.Storage;
+using Windows.Graphics.Imaging;
+using Windows.Media;
+using Microsoft.AI.MachineLearning;
+
+namespace WinMLSamplesGallery.Samples
+{
+    public sealed class EvalResult
+    {
+        public string nonBatchedAvgTime { get; set; }
+        public string batchedAvgTime { get; set; }
+        public string timeRatio { get; set; }
+    }
+
+    public sealed partial class Batching : Page
+    {
+        private LearningModelSession nonBatchingSession_;
+        private LearningModelSession BatchingSession_;
+        float avgNonBatchedDuration = 0;
+        float avgBatchDuration = 0;
+        static bool navigatingAwayFromPage = false;
+
+        public Batching()
+        {
+            this.InitializeComponent();
+            // Ensure static variable is always false on page initialization
+            navigatingAwayFromPage = false;
+        }
+
+        async private void StartInference(object sender, RoutedEventArgs e)
+        {
+            ShowEvalUI();
+            ResetEvalMetrics();
+
+            var inputImages = GetInputImages();
+            int batchSize = GetBatchSizeFromBatchSizeComboBox();
+            CreateSessions(batchSize);
+
+            UpdateEvalText(false);
+            await Classify(inputImages);
+
+            UpdateEvalText(true);
+            await ClassifyBatched(inputImages, batchSize);
+
+            GenerateEvalResultAndUI();
+        }
+
+        private void ShowEvalUI()
+        {
+            StartInferenceBtn.IsEnabled = false;
+            EvalResults.Visibility = Visibility.Collapsed;
+            LoadingContainer.Visibility = Visibility.Visible;
+        }
+
+        private void ResetEvalMetrics()
+        {
+            avgNonBatchedDuration = 0;
+            avgBatchDuration = 0;
+        }
+
+        private List<VideoFrame> GetInputImages()
+        {
+            var birdFile = StorageFile.GetFileFromApplicationUriAsync(new Uri("ms-appx:///InputData/hummingbird.jpg")).GetAwaiter().GetResult();
+            var catFile = StorageFile.GetFileFromApplicationUriAsync(new Uri("ms-appx:///InputData/kitten.png")).GetAwaiter().GetResult();
+            var birdImage = CreateSoftwareBitmapFromStorageFile(birdFile);
+            var catImage = CreateSoftwareBitmapFromStorageFile(catFile);
+            var inputImages = new List<VideoFrame>();
+            for (int i = 0; i < 25; i++)
+            {
+                inputImages.Add(VideoFrame.CreateWithSoftwareBitmap(birdImage));
+                inputImages.Add(VideoFrame.CreateWithSoftwareBitmap(catImage));
+            }
+            return inputImages;
+        }
+
+        private SoftwareBitmap CreateSoftwareBitmapFromStorageFile(StorageFile file)
+        {
+            var stream = file.OpenAsync(FileAccessMode.Read).GetAwaiter().GetResult();
```

Comment:
can you not just use "await"? Instead of var stream = file.OpenAsync(FileAccessMode.Read).GetAwaiter().GetResult();, you can do var stream = await file.OpenAsync(FileAccessMode.Read);
there's a few more instances of this in this file too

Reply:
It's not equivalent, he doesn't want `CreateSoftwareBitmapFromStorageFile` to be async, so he can't use await. As to why/if it's a good decision I'll let him justify his choice

Reply:
I have no particular preference. This was just a function I copied from @smk2007 Image Classifier sample

Reply:
If you can convert the method to async and use await I'd recommend doing so, as GetAwaiter() (or Task.Run()) basically is calling async method synchronously and has a potential to cause deadlock.

Reply:
Done.